repo-agent/issue-sandbox/main.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+)
+
+func main() {
+	cmdCodeSrv, err := startCodeServer()
+	if err != nil {
+		log.Fatalf("failed to start code-server: %v", err)
+	}
+	defer func() {
+		if cmdCodeSrv.Process != nil {
+			_ = cmdCodeSrv.Process.Kill()
+		}
+	}()
+
+	err = runIssueSolver()
+	if err != nil {
+		log.Fatalf("failed solving issue: %v", err)
+	}
+
+	err = cmdCodeSrv.Wait()
+	if err != nil {
+		log.Printf("Code Server exited with error: %v", err)
+	} else {
+		log.Println("Code Server exited with no error")
+	}
+}
+
+func runIssueSolver() error {
+	log.Println("Starting issue solver")
+
+	// Environment variables
+	gitPushEnabled := os.Getenv("GIT_PUSH_ENABLED") == "true"
+	githubUserOrigin := os.Getenv("GITHUB_USER_ORIGIN")
+	githubUserLogin := os.Getenv("GITHUB_USER_LOGIN")
+	githubToken := os.Getenv("GITHUB_TOKEN")
+	githubUserEmail := os.Getenv("GITHUB_USER_EMAIL")
+	githubUserName := os.Getenv("GITHUB_USER_NAME")
+	issueBranch := os.Getenv("ISSUE_BRANCH")
+	agentPrompt := os.Getenv("AGENT_PROMPT")
+	issueID := os.Getenv("ISSUEID")
+
+	if _, err := runCommand("git", "remote", "remove", "origin"); err != nil {
+		log.Printf("could not remove origin, probably because it does not exist: %v", err)
+	}
+
+	if gitPushEnabled && githubUserOrigin != "" {
+		originURL := fmt.Sprintf("https://%s:%s@%s", githubUserLogin, githubToken, githubUserOrigin)
+		if _, err := runCommand("git", "remote", "add", "origin", originURL); err != nil {
+			return fmt.Errorf("failed to add origin: %w", err)
+		}
+	}
+
+	if githubUserEmail != "" {
+		if _, err := runCommand("git", "config", "--global", "user.email", githubUserEmail); err != nil {
+			return fmt.Errorf("failed to set git user email: %w", err)
+		}
+	}
+
+	if githubUserName != "" {
+		if _, err := runCommand("git", "config", "--global", "user.name", githubUserName); err != nil {
+			return fmt.Errorf("failed to set git user name: %w", err)
+		}
+	}
+
+	oldCommitID, err := runCommand("git", "rev-parse", "HEAD")
+	if err != nil {
+		return fmt.Errorf("failed to get old commit id: %w", err)
```

Comment:
The `git checkout -b` command will fail if the branch specified by `ISSUE_BRANCH` already exists. To prevent this, consider using `git checkout -B` which creates the branch if it doesn't exist or resets it to the current commit if it does. This would make the process more resilient.